### PR TITLE
Able to run menu commands while in commandline

### LINKED ISSUE #3386: :emenu not executing command mode commands
I am sending the command  `emenu Edit.Paste` while in command mode and instead of getting the text that is in the clipboard as is done  when pasting in both normal and insert mode, it just pastes `"+gP^`.  Am I supposed to be doing anything different to get the command mode menu to work?

Doing `:amenu Edit.Paste` shows that it should do ` <C-R>+`
```vim
360 &Paste^I"+gP
    n*   "+gP
    v&   "-c<Esc>:call paste#Paste()<CR>
    s&   "-c<Esc>:call paste#Paste()<CR>
    i&   x<BS><Esc>:call paste#Paste()<CR>gi
    c*   <C-R>+
````
However, doing `:menu Edit.Paste` does not show the `<C-R>+`:
````vim
360 &Paste^I"+gP
    n*   "+gP
    v&   "-c<Esc>:call paste#Paste()<CR>
    s&   "-c<Esc>:call paste#Paste()<CR>
```

Here's a test that replicates the issue. It should work being inserted into the `input_spec.lua`. The test loads menu.vim, inserts some text and copies the first word, pastes it to make sure that paste works, goes into command mode and tries to paste again. 

```lua
describe('Paste in commandline', function()
    it('ok', function()
        local screen = require('test.functional.ui.screen').new()
        screen:attach()
        nvim('command', 'runtime menu.vim')
        feed('ithis is a sentence<esc>^"+yiwo<esc>')
        nvim('command', 'emenu Edit.Paste')
        feed(':')
        nvim('command', 'emenu Edit.Paste')
        screen:snapshot_util({}, true)
        screen:expect([[
          this is a sentence                                   |
          this                                                 |
          ~                                                    |
          ~                                                    |
          ~                                                    |
          ~                                                    |
          ~                                                    |
          ~                                                    |
          ~                                                    |
          ~                                                    |
          ~                                                    |
          ~                                                    |
          ~                                                    |
          :this^                                                |
        ]])
    end)
end)
```

### PR DESCRIPTION
**Pull request by bambu**
Running `:emenu` while in commandline does not run the `cmenu` mapping, it runs the normal mode mapping instead. Added a check to the state to make set the `MENU_INDEX_CMDLINE` when in the CMDLINE state. 

This should resolve #3386.

### CODE REVIEW

File: src/nvim/menu.c [hunk 1]
```diff
@@ -1308,6 +1308,9 @@ void ex_emenu(exarg_T *eap)
      * for exclusive mode */
     if (*p_sel == 'e' && gchar_cursor() != NUL)
       ++curwin->w_cursor.col;
+  } else if((State & CMDLINE)) {
```
Comment (justinmk) [inline src/nvim/menu.c]: can you remove the extra parens? clint will also complain about the lack of a space in `if(`.

**Comment by justinmk** [comment]
Can you include the test as well?

File: src/nvim/menu.c [hunk 2]
```diff
@@ -1308,6 +1308,9 @@ void ex_emenu(exarg_T *eap)
      * for exclusive mode */
     if (*p_sel == 'e' && gchar_cursor() != NUL)
       ++curwin->w_cursor.col;
+  } else if ((State & CMDLINE)) {
```
Comment (justinmk) [inline src/nvim/menu.c]: remove extra parens

**Comment by bambu** [comment]
Yeah I can include the test. Should I leave it in input_spec or have it be in its own file?

**Comment by justinmk** [comment]
`test/functional/ex_cmds/menu_spec.lua` already exists, it seems like a good place.

File: src/nvim/menu.c [hunk 3]
```diff
@@ -1308,6 +1308,9 @@ void ex_emenu(exarg_T *eap)
      * for exclusive mode */
     if (*p_sel == 'e' && gchar_cursor() != NUL)
       ++curwin->w_cursor.col;
+  } else if (State & CMDLINE) {
```
Comment (justinmk) [inline src/nvim/menu.c]: What happens if `eap->addr_count` is non-zero while `(State & CMDLINE)`? Is that possible? Does line 1265 also need to be changed to `else if (!(State & CMDLINE) && eap->addr_count)` or is it ok for `mode` to be `"Visual"` in that case?

File: test/functional/ex_cmds/menu_spec.lua [hunk 4]
```diff
@@ -36,3 +36,37 @@ describe(':emenu', function()
   end)
 
 end)
+
+describe('emenu Edit.Paste while in commandline', function()
+    before_each(clear)
+
+    it('ok', function()
+        local screen = require('test.functional.ui.screen').new()
```
Comment (justinmk) [inline test/functional/ex_cmds/menu_spec.lua]: Can you change the `new()` call to `new(40, 8)` and adjust the test below? Most tests (like this one) don't need such a big space, so this compacts the tests a bit for readability.

**Comment by bambu** [comment]
Made the requested change to the test. 

I could not get the CMDLINE if statement to fail even in visual mode. I changed the location of the if statement.

**Comment by justinmk** [comment]
Merged after squash.

**Pull request closed without merging**